### PR TITLE
Tab Home

### PR DESCRIPTION
**Pull request by HaVuNguyen**

### CODE REVIEW

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -1,7 +1,233 @@
...
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     tools:context=".MainActivity">
-</android.support.constraint.ConstraintLayout>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="90dp"
+        android:background="#5A6978"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: hard code

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -1,7 +1,233 @@
...
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     tools:context=".MainActivity">
-</android.support.constraint.ConstraintLayout>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="90dp"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: hard code

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -1,7 +1,233 @@
...
+        android:background="#5A6978"
+        android:orientation="horizontal">
+
+        <TextView
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:padding="10dp"
+            android:text="HOME"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: hard code

File: app/src/main/res/drawable/ic_acoustic.xml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?><?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:width="33.003dp"
+    android:height="33.003dp"
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_acoustic.xml]: hard code

File: app/src/main/res/drawable/ic_drummer.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:width="19.099dp"
+    android:height="19.099dp"
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_drummer.xml]: hard code

File: app/src/main/res/drawable/ic_headphones.xml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:width="480.058dp"
+    android:height="480.058dp"
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_headphones.xml]: hard code

File: app/src/main/res/drawable/ic_headphones.xml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+        tools:ignore="VectorPath" />
+</vector>
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_headphones.xml]: enter end line

File: app/src/main/res/drawable/ic_home.xml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:width="511.999dp"
+    android:height="511.999dp"
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_home.xml]: hard code

File: app/src/main/res/drawable/ic_home.xml [hunk 9]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_home.xml]: remove this line

File: app/src/main/res/drawable/ic_music.xml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="54dp"
+    android:height="54dp"
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_music.xml]: hard code

File: app/src/main/res/drawable/ic_person.xml [hunk 11]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:width="612dp"
+    android:height="612dp"
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_person.xml]: hard code

File: app/src/main/res/drawable/ic_piano.xml [hunk 12]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="360.08dp"
+    android:height="360.08dp"
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_piano.xml]: hard code

File: app/src/main/res/drawable/ic_saxaphone.xml [hunk 13]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:width="512dp"
+    android:height="512dp"
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_saxaphone.xml]: hard code

File: app/src/main/res/drawable/ic_search.xml [hunk 14]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="512dp"
+    android:height="512dp"
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_search.xml]: hard code

File: app/src/main/res/layout-v21/activity_main.xml [hunk 15]
```diff
@@ -0,0 +1,245 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/dp_100"
+        android:background="@color/color_background"
+        android:orientation="horizontal">
+
+        <TextView
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:padding="@dimen/dp_10"
+            android:text="HOME"
```
Comment (Tuanlvt) [inline app/src/main/res/layout-v21/activity_main.xml]: hard code

File: app/src/main/res/layout-v21/activity_main.xml [hunk 16]
```diff
@@ -0,0 +1,245 @@
...
+            android:textAlignment="center"
+            android:textColor="@color/color_while"
```
Comment (Tuanlvt) [inline app/src/main/res/layout-v21/activity_main.xml]: change to @android:color/white

File: app/src/main/res/layout-v21/activity_main.xml [hunk 17]
```diff
@@ -0,0 +1,245 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (Tuanlvt) [inline app/src/main/res/layout-v21/activity_main.xml]: change to ConstraintLayout

File: app/src/main/res/layout/activity_main.xml [hunk 18]
```diff
@@ -1,7 +1,245 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: change to ConstraintLayout

File: app/src/main/res/values/colors.xml [hunk 19]
```diff
@@ -3,4 +3,8 @@
     <color name="colorPrimary">#3F51B5</color>
     <color name="colorPrimaryDark">#303F9F</color>
     <color name="colorAccent">#FF4081</color>
+    <color name="color_black">#000</color>
+    <color name="color_background">#5A6978</color>
+    <color name="color_while">#fff</color>
+    <color name="color_vangnhat">#FFD185</color>
```
Comment (Tuanlvt) [inline app/src/main/res/values/colors.xml]: pls create name is English

File: app/src/main/res/layout/activity_main.xml [hunk 20]
```diff
@@ -3,5 +3,172 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    tools:context=".MainActivity">
+    tools:layout_editor_absoluteY="89dp">
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: hard code

File: app/src/main/res/layout-v21/activity_main.xml [hunk 21]
```diff
@@ -0,0 +1,174 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:layout_editor_absoluteY="89dp">
```
Comment (Tuanlvt) [inline app/src/main/res/layout-v21/activity_main.xml]: hard code

File: app/src/main/res/drawable/ic_drummer.xml [hunk 22]
```diff
@@ -0,0 +1,10 @@
...
+</vector>
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_drummer.xml]: enter end line

File: app/src/main/res/drawable/ic_headphones.xml [hunk 23]
```diff
@@ -0,0 +1,12 @@
...
```
Comment (Tuanlvt) [inline app/src/main/res/drawable/ic_headphones.xml]: enter end line

File: app/src/main/res/layout-v21/activity_main.xml [hunk 24]
```diff
@@ -0,0 +1,172 @@
...
+    tools:context=".MainActivity">
+    <view
+        class="android.support.constraint.ConstraintLayout"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/dp_100"
+        android:background="@color/color_background"
+        tools:ignore="MissingConstraints"
+        tools:layout_editor_absoluteX="0dp" />
+    <TextView
+        android:id="@+id/textView"
```
Comment (Tuanlvt) [inline app/src/main/res/layout-v21/activity_main.xml]: check convention

File: app/src/main/res/layout-v21/activity_main.xml [hunk 25]
```diff
@@ -0,0 +1,172 @@
...
+        tools:layout_editor_absoluteX="230dp"
+        tools:layout_editor_absoluteY="16dp" />
+    <view
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/dp_100"
+        tools:ignore="MissingConstraints"
+        tools:layout_editor_absoluteX="0dp"
+        tools:layout_editor_absoluteY="723dp" />
+    <Button
+        android:id="@+id/btn_home"
```
Comment (Tuanlvt) [inline app/src/main/res/layout-v21/activity_main.xml]: check convention

File: app/src/main/res/layout/activity_main.xml [hunk 26]
```diff
@@ -1,7 +1,212 @@
...
+        app:layout_constraintTop_toTopOf="parent"
+        />
+    <android.support.constraint.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_margin="@dimen/sp_50"
+        android:layout_marginTop="@dimen/dp_40"
+        app:layout_constraintTop_toBottomOf="@+id/text_home"
+        tools:ignore="MissingConstraints"
+        tools:layout_editor_absoluteX="50dp"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: remove line (30,31)

File: app/src/main/res/layout/activity_main.xml [hunk 27]
```diff
@@ -1,7 +1,212 @@
...
+        android:textSize="@dimen/sp_50"
+        android:textStyle="bold"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        />
+    <android.support.constraint.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_margin="@dimen/sp_50"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: (tat) what sp?

File: app/src/main/res/layout/activity_main.xml [hunk 28]
```diff
@@ -1,7 +1,212 @@
...
+        android:layout_marginTop="@dimen/dp_40"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: remove this line

File: app/src/main/res/layout/activity_main.xml [hunk 29]
```diff
@@ -1,7 +1,212 @@
...
+            />
+    </android.support.constraint.ConstraintLayout>
+    <android.support.constraint.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="84dp"
+        android:background="@color/color_background"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/text_nhac_acoustic"
+        tools:ignore="MissingConstraints,NotSibling"
+        tools:layout_editor_absoluteX="0dp"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: remove (181, 182)

File: app/src/main/res/layout/activity_main.xml [hunk 30]
```diff
@@ -1,7 +1,212 @@
...
+            android:textColor="@color/color_black"
+            android:textSize="@dimen/sp_30"
+            android:textStyle="bold"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/button_nhac_han_quoc"
+            />
+    </android.support.constraint.ConstraintLayout>
+    <android.support.constraint.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="84dp"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: hard code

File: app/src/main/res/values/colors.xml [hunk 31]
```diff
@@ -3,4 +3,8 @@
     <color name="colorPrimary">#3F51B5</color>
     <color name="colorPrimaryDark">#303F9F</color>
     <color name="colorAccent">#FF4081</color>
+    <color name="color_black">#000000</color>
+    <color name="color_background">#5A6978</color>
+    <color name="color_white">#FFFFFF</color>
```
Comment (Tuanlvt) [inline app/src/main/res/values/colors.xml]: remove if not used

File: app/src/main/res/values/dimens.xml [hunk 32]
```diff
@@ -0,0 +1,24 @@
...
+    <dimen name="dp_133">133dp</dimen>
+    <dimen name="dp_80">80dp</dimen>
+    <dimen name="dp_55">55dp</dimen>
+    <dimen name="dp_20">20dp</dimen>
+    <dimen name="dp_70">70dp</dimen>
+    <dimen name="dp_40">40dp</dimen>
+    <dimen name="dp_720">720dp</dimen>
+    <dimen name="dp_54">54dp</dimen>
+    <dimen name="dp_612">612dp</dimen>
+    <dimen name="dp_512">512dp</dimen>
```
Comment (Tuanlvt) [inline app/src/main/res/values/dimens.xml]: remove if not used

File: app/src/main/res/layout/activity_main.xml [hunk 33]
```diff
@@ -1,7 +1,209 @@
...
+            android:layout_marginStart="@dimen/dp_70"
+            android:background="@drawable/ic_piano"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintLeft_toRightOf="@+id/button_all_music"
+            app:layout_constraintTop_toTopOf="parent"
+            />
+        <TextView
+            android:id="@+id/text_nhac_tre"
+            android:layout_width="wrap_content"
+            android:layout_height="39dp"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: hard code

File: app/src/main/res/values/dimens.xml [hunk 34]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="sp_30">30sp</dimen>
+    <dimen name="sp_50">50sp</dimen>
```
Comment (Tuanlvt) [inline app/src/main/res/values/dimens.xml]: remove if not used

File: app/src/main/res/layout/activity_main.xml [hunk 35]
```diff
@@ -1,7 +1,209 @@
 <?xml version="1.0" encoding="utf-8"?>
 <android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    tools:context=".MainActivity">
+    tools:context=".MainActivity"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: remove

File: app/src/main/AndroidManifest.xml [hunk 36]
```diff
@@ -1,18 +1,22 @@
...
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_square"
         android:supportsRtl="true"
-        android:theme="@style/AppTheme">
-        <activity android:name=".MainActivity">
+        android:theme="@style/AppTheme"
+        tools:ignore="AllowBackup,GoogleAppIndexingWarning">
+        <activity
+            android:name=".MainActivity"
+            android:theme="@style/AppTheme">
```
Comment (Tuanlvt) [inline app/src/main/AndroidManifest.xml]: add screenOrientation

File: app/src/main/java/com/framgia/music_09/MainActivity.java [hunk 37]
```diff
@@ -1,7 +1,7 @@
 package com.framgia.music_09;
 
-import android.os.Bundle;
-import android.support.v7.app.AppCompatActivity;
+        import android.os.Bundle;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/music_09/MainActivity.java]: format code

File: app/src/main/res/layout/activity_main.xml [hunk 38]
```diff
@@ -1,7 +1,152 @@
...
+        />
+    <android.support.constraint.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/dp_84"
+        android:background="@color/color_background"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintVertical_bias="1.0"
+        tools:ignore="MissingConstraints,NotSibling"
```
Comment (Tuanlvt) [inline app/src/main/res/layout/activity_main.xml]: remove it

File: app/src/main/java/com/framgia/music_09/MainActivity.java [hunk 39]
```diff
@@ -1,7 +1,7 @@
...
+        import android.support.v7.app.AppCompatActivity;
```
Comment (Tuanlvt) [inline app/src/main/java/com/framgia/music_09/MainActivity.java]: reformat code

File: app/src/main/res/values/strings.xml [hunk 40]
```diff
@@ -1,3 +1,10 @@
 <resources>
     <string name="app_name">iMusic</string>
+    <string name="music_acoustic">Nhạc Acoustic</string>
+    <string name="music_hanquoc">Nhạc Hàn Quốc</string>
+    <string name="music_aumy">Nhạc Âu Mỹ</string>
+    <string name="music_edm">Nhạc EDM</string>
+    <string name="music_all">All Music</string>
+    <string name="music_nhactre">Nhạc trẻ</string>
```
Comment (ThuanPx) [inline app/src/main/res/values/strings.xml]: music_nhactre -> tiếng anh đi e

**Pull request merged**